cloud/amazon/ec2_ami.py

```diff
@@ -87,7 +87,68 @@
     required: false
     default: null
     version_added: "2.0"
-author: "Evan Duffield (@scicoin-project) <eduffield@iacquire.com>"
+  image_location:
+    description:
+      - Full path to your AMI manifest in Amazon S3 storage. Only used for S3-based AMI's.
+    required: false
+    default: null
+    version_added: "2.0"
+  action:
+    description:
+      - Whether the ami should be created or registered. 
+    choices: [ "create", "register" ]
+    required: true 
+    default: create
+    version_added: "2.0"
+  architecture:
+    description:
+      - The architecture of the AMI. 
+    choices: [ "i386", "x86_64" ]
+    required: true 
+    default: x86_64
+    version_added: "2.0"
+  kernel_id:
+    description:
+      - The ID of the kernel with which to launch the instances
+    required: false
+    default: null
+    version_added: "2.0"
+  root_device_name:
+    description:
+      - The root device name (e.g. /dev/sda1)
+    required: false
+    default: null
+    version_added: "2.0"
+  virtualization_type:
+    description:
+      - The virutalization_type of the image.
+    choices: [ "hvm", "paravirtual" ]
+    required: true
+    default: hvm
+    version_added: "2.0"
+  sriov_net_support:
+    description:
+      - Advanced networking support.
```

Comment:
Be more descriptive, it's not clear what the default behavior is, my options are no advanced networking support, and simple advanced networking support?